### PR TITLE
JOSS paper

### PR DESCRIPTION
**Pull request by SimonBiggs**
Hi All,

@rembishj has put together a great draft top level paper for citing PyMedPhys. This PR brings in his work. The current authors on the paper are:

@SimonBiggs, @sjswerdloff, @Matthew-Jennings, @pchlap, and @rembishj. But there have been many others involved in the formation during these early years of PyMedPhys:

@msobolewski, @JacobMcAloney, @dg1an3, @matthewdeancooper, @kingrpaul, @peterg1t, and @ayalalazaro.

Given your involvement in being a part of making PyMedPhys what it is today, if you'd like to be involved in the authorship of this paper, let us know in this PR.

Everyone who is an author, can you please review this PR, make comments, suggestions, and if you're happy with it provide your PR approval.

Thanks again @rembishj for making the writing of this paper a reality!

Cheers,
Simon

### CODE REVIEW

**Review by sjswerdloff — APPROVED** [review]

**Comment by SimonBiggs** [comment]
Here are the details regarding authorship requirements:

https://joss.readthedocs.io/en/latest/submitting.html#authorship

Also, before we submit this for review, here is the review checklist and review criteria:

- https://joss.readthedocs.io/en/latest/review_checklist.html
- https://joss.readthedocs.io/en/latest/review_criteria.html

**Comment by SimonBiggs** [comment]
Compiled paper pdf can be downloaded from:

https://github.com/pymedphys/pymedphys/actions/runs/2008055087

**Review by pchlap — APPROVED** [review]
This reads really well. I've left a very minor comment which you can take or leave as you like. Great work!

**Review by rembishj — APPROVED** [review]
Thanks for looking over it!

File: paper.md
Comment (Matthew-Jennings) [inline paper.md:61]: perhaps *"PyMedPhys aims to improve the quality and accessibility of existing software solutions to problems faced by medical physicists"*?

I think it makes sense to focus on *medical physics-related applications* rather than *medical physicists*, noting that plenty of PyMedPhys' code was written by *non*-medical physicists, nor is it only useful to medical physicists.

File: paper.md
Comment (Matthew-Jennings) [inline paper.md:62]:
```suggestion
PyMedPhys is currently implemented around the world for a handful of
```

File: paper.md
Comment (Matthew-Jennings) [inline paper.md:63]: Suggest "*Many parties have embraced the gamma analysis module [refs] and implementations of the electron cutout factor module [refs], *y* module [refs], [...], and *z* module [refs] have also been reported.*

**Review by Matthew-Jennings — CHANGES_REQUESTED** [review]
Thanks very much, @rembishj, for getting this going! Great start :)

**Comment by Matthew-Jennings** [comment]
I haven't really looked into JOSS before. What are everyone's thoughts on the scope of this kind of top level paper? E.g., should it include some module details?

**Comment by SimonBiggs** [comment]
> What are everyone's thoughts on the scope of this kind of top level paper?

Here is what the paper should contain:

https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain

**Review by rembishj — APPROVED** [review]

**Comment by SimonBiggs** [comment]
@Matthew-Jennings have those changes that you requested been met?

**Comment by SimonBiggs** [comment]
@rembishj I would have thought some of the other contributors would have been interested in being authors. But they haven't raised their voice here... Nevertheless, I think within the acknowledgement section we should list by name the following past contributors:

@msobolewski, @JacobMcAloney, @dg1an3, @matthewdeancooper, @kingrpaul, @peterg1t, and @ayalalazaro.

**Comment by JacobMcAloney** [comment]
@SimonBiggs and @rembishj 

Sorry I haven't replied till now. I am more than happy (keen even) to be involved in the publication, the only thing being I don't feel like I could contribute meaningfully to the real meat of the work. That said, if you want proof reading or suggested edits, or even overview sections drafted I'd love to be involved.

**Comment by SimonBiggs** [comment]
> I am more than happy (keen even) to be involved in the publication

Awesome :) (and... I still need to reply about path finder :) )

Here's a link to the latest rendering of the paper draft:

https://github.com/pymedphys/pymedphys/suites/5793995343/artifacts/193441134

Here are the details of what is required for authorship:

> Purely financial (such as being named on an award) and organizational (such as general supervision of a research group) contributions are not considered sufficient for co-authorship of JOSS submissions, but active project direction and other forms of non-code contributions are. The authors themselves assume responsibility for deciding who should be credited with co-authorship, and co-authors must always agree to be listed. In addition, co-authors agree to be accountable for all aspects of the work, and to notify JOSS if any retraction or correction of mistakes are needed after publication.

https://joss.readthedocs.io/en/latest/submitting.html#authorship

For large swathes of PyMedPhys you fulfilled the following through guidance, feedback, validation, testing, etc:

> but active project direction and other forms of non-code contributions are

(And even a code contribution here and there :slightly_smiling_face: )

So it'd be an honour to be able to have you on as an author (@msobolewski too if he's keen).

At this point in time all you need to do is review the paper linked above and mark your approval on this pull request :slightly_smiling_face:. If you're keen you could even jump over to the file over at:

https://github.com/pymedphys/pymedphys/pull/1622/files#diff-0403ef06adf405f7b310b4518bd6a3559854f54c61676f676ce9cbfee7172ab6

And use the following steps to add your name to the author list:

image

**Review by JacobMcAloney — APPROVED** [review]

**Comment by SimonBiggs** [comment]
@rembishj I removed the "co-first author" footnotes. Is that okay?

**Comment by ayalalazaro** [comment]
Hi all,
I am sorry I didn t reply before, for some reason I think I didn t receive a notification the first time @SimonBiggs tagged me.
I think I am in a similar situation than @JacobMcAloney. My contribution to pymedphys was small and I am not sure if I can contribute in a substantial way to the project. That being said, I am more than interested in collaborating reviewing, testing or performing any task that you will find suitable for me. 
Thank you for counting on me,
Rafael

**Comment by SimonBiggs** [comment]
Hi @ayalalazaro,

> My contribution to pymedphys was small and I am not sure if I can contribute in a substantial way to the project.

It was really great to be able to have [your contribution](https://github.com/pymedphys/pymedphys/pull/870) to the "labs" way back when :)

If you are interested in having ongoing responsibility for that part of the code base that you contributed:

* https://github.com/pymedphys/pymedphys/blob/be980e9f92232d612865107b63a15012e6afeca1/lib/pymedphys/_experimental/quickcheck/qcheck.py
* https://github.com/pymedphys/pymedphys/blob/be980e9f92232d612865107b63a15012e6afeca1/lib/pymedphys/_experimental/quickcheck/quickcheck_cli.py

Where "ongoing responsibility" entails answering any questions that people have when they try and use that code you wrote, contributing or reviewing bug fixes to that portion of the code base, and at some point being willing to write some automated tests for it to make sure that the code still works, then I see no reason why you can't be an author on this PyMedPhys paper :slightly_smiling_face:.

**Comment by kingrpaul** [comment]
> @rembishj I would have thought some of the other contributors would have been interested in being authors. But they haven't raised their voice here... Nevertheless, I think within the acknowledgement section we should list by name the following past contributors:
> 
> @msobolewski, @JacobMcAloney, @dg1an3, @matthewdeancooper, @kingrpaul, @peterg1t, and @ayalalazaro.

Sorry for being slow to reply. Happy to help any way I can. Just let me know. - Paul

**Comment by SimonBiggs** [comment]
> Sorry for being slow to reply.

All good, no stress. Hope you're doing well, staying healthy :).

Would you be interested in being included as an author of the paper? Or would you prefer an acknowledgement at the end?

**Comment by kingrpaul** [comment]
> > Sorry for being slow to reply.
> 
> All good, no stress. Hope you're doing well, staying healthy :).
> 
> Would you be interested in being included as an author of the paper? Or would you prefer an acknowledgement at the end?

Happy to be an author if appropriate and I can contribute. I've been out of the loop for a while. Stayed healthy and (mostly) out of trouble. Hope the same is true for you. Catch me up, if you would. What's  going on?

**Comment by SimonBiggs** [comment]
> Happy to be an author if appropriate and I can contribute.

Awesome, chuck your name in after Jacob and before Matt utilising the edit function over at https://github.com/pymedphys/pymedphys/pull/1622/files#diff-0403ef06adf405f7b310b4518bd6a3559854f54c61676f676ce9cbfee7172ab6:

image

> Catch me up, if you would. What's going on?

The community has grown. The maintainer count now sits at 4, with the help from @sjswerdloff, @Matthew-Jennings, and @pchlap. That has been amazing to be able to spread that workload around. This is specifically important since I am no longer working in the clinic, but instead, 10 months ago, created my own AI startup with the first product being autocontouring (which, in early trials surpasses all the competition :tada:).

PyMedPhys itself has come along in leaps and bounds. Some of the parts I am most proud of are the Elekta Icom streamer, the TRF logfile recorder, and the MU Density. Which have all been combined into a logfile analyser which is still in regular use at my old work and at some centres around the world. Another component which was great was the WLutz Arc which is able to measure the beam position over every gantry angle in 1 minute. My old work is still utilising that within their daily warm up procedure without costing the treatment staff any extra time (they use the warm up beams to automatically deliver arcs getting out fine grained gantry and energy beam position data, every day). A shout out needs to go to @sjswerdloff, who put up with my barrage of PR review requests through the creation of that WLutz Arc and others :slightly_smiling_face:.

We had some great projects come through. @matthewdeancooper with his masters for an AI project, and @rembishj with his PhD implementing TG275.

@pchlap brought in some tools they'd been utilising to export pinnacle databases on bulk. @dg1an3 from Elekta provided some great work around Mosaiq database querying. @Matthew-Jennings and @sjswerdloff pulled in tools around DICOM anonymisation which have been immensely helpful to many.

But really... to get a bit of a handle on all that has changed, it might be worth having a look through the release notes (https://docs.pymedphys.com/release-notes.html). That'll give a pretty good idea of what has been happening :slightly_smiling_face:.

**Comment by ayalalazaro** [comment]
> Hi @ayalalazaro,
> 
> > My contribution to pymedphys was small and I am not sure if I can contribute in a substantial way to the project.
> 
> It was really great to be able to have [your contribution](https://github.com/pymedphys/pymedphys/pull/870) to the "labs" way back when :)
> 
> If you are interested in having ongoing responsibility for that part of the code base that you contributed:
> 
>     * https://github.com/pymedphys/pymedphys/blob/be980e9f92232d612865107b63a15012e6afeca1/lib/pymedphys/_experimental/quickcheck/qcheck.py
> 
>     * https://github.com/pymedphys/pymedphys/blob/be980e9f92232d612865107b63a15012e6afeca1/lib/pymedphys/_experimental/quickcheck/quickcheck_cli.py
> 
> 
> Where "ongoing responsibility" entails answering any questions that people have when they try and use that code you wrote, contributing or reviewing bug fixes to that portion of the code base, and at some point being willing to write some automated tests for it to make sure that the code still works, then I see no reason why you can't be an author on this PyMedPhys paper 🙂.

Sure, I am going to write those tests, thanks!

**Review by ayalalazaro — APPROVED** [review]
reviewed

**Comment by SimonBiggs** [comment]
Sorry all about the multiple "auto merges" that are likely filling up everyone's inboxes. The pre-commit fixer bot was playing up and going into a self perpetuating loop. ... Hopefully fixed now.

**Review by kingrpaul — APPROVED** [review]
Looks good.

File: paper.md
Comment (dg1an3) [inline paper.md:28]: should I add my name and affiliation if I want to be on the authors list?

Comment (SimonBiggs) [reply paper.md:28]: Yes please! :) Thanks Derek :)

Comment (SimonBiggs) [reply paper.md:28]: Put your name in between the two Jacobs:

image

And then just tidy up the affiliation numbering and ordering so that it is still in order once you also put the affiliation in.

**Comment by SimonBiggs** [comment]
Would there be value in having this paper reference as well as acknowledge the pre-existing python medical physics ecosystem? (pydicom, dicompyler, pylinac, qatrackplus)

* @darcymason, @mrbean-bremen, @scaramallion?
* @bastula?
* @jrkerns?
* @randlet?

Would there be value in making reference to other parts of the ecosystem that has also come into existence recently such as DVH Analytics (@cutright), and OnkoDICOM (@didymo, @sjswerdloff)?

Are there some other core parts of the python medical physics ecosystem that I've missed?

**Comment by sjswerdloff** [comment]
IMHO, Mention of the imports and inspiration like pydicom and pylinac, yes.
Additional consumers like OnkoDICOM maybe not, unless they have already asked for how to reference us.

**Comment by cutright** [comment]
Thanks @SimonBiggs, I appreciate the mention and consideration, but I'm not sure a reference to DVHA fits. Unless some code was borrowed? Besides... its old technology at this point anyway. Can't wait to release the next generation  =)

However, I think a reference to @bastula and `dicompyler-core` makes sense, which appears to be explicitly used (and it is a very valuable library, for the radonc side of the field at least).

**Comment by darcymason** [comment]
> * @darcymason, @mrbean-bremen, @scaramallion? (https://doi.org/10.1118/1.3611983)

An acknowledgement would be nice, but its up to your authors.  I don't think a reference is necessary, but if that is needed, then as per ["How do I cite pydicom"](https://pydicom.github.io/pydicom/dev/faq/index.html#how-do-i-cite-pydicom) you could just use the DOI to a recent version, or the reference you mentioned can also work.

**Comment by bastula** [comment]
> * @bastula? (http://dx.doi.org/10.1118/1.3468652)

Thanks @SimonBiggs for the acknowledgement and possible inclusion. Either citing `dicompyler` as a pre-existing project (using the DOI you listed above) or `dicompyler-core` as ongoing work would be fine. For `dicompyler-core` you can use the latest version DOI: https://dx.doi.org/10.5281/zenodo.3236628

**Comment by ayalalazaro** [comment]
> Are there some other core parts of the python medical physics ecosystem that I've missed?

My two cents on the python medical physics ecosystem:
I think pyQAserver is a very nice piece of software too, it extends pylinac and connects to Orthanc PACS (which is a awesome too!)
https://github.com/brjdenis/pyqaserver
https://www.orthanc-server.com/index.php

Dennis Brojan' s tips for Elekta users are really useful, definately worth reading.
https://synergyqatips.readthedocs.io/en/latest/

**Comment by matthewdeancooper** [comment]
Hey guys, been MIA from github as im not posting public code lately. I'll have to check with work if its ok to contribute - i dont think there is any conflict but its something i should check before i raise my hand for future work. Its unlikely this will be in any substantial way as I'm pretty short on time these days. That being said, I'm happy to review the paper and be included if you think my past contributions warrant it?

**Comment by SimonBiggs** [comment]
> I'm happy to review the paper and be included if you think my past contributions warrant it?

It for sure does :slightly_smiling_face:.

**Comment by SimonBiggs** [comment]
> Dennis Brojan' s tips for Elekta users are really useful, definitely worth reading.
> https://synergyqatips.readthedocs.io/en/latest/

These tips are amazing. @JacobMcAloney you might like having a look (specifically the FFF focal spot stuff :slightly_smiling_face:)

**Comment by dg1an3** [comment]
@SimonBiggs I double-checked the affiliations after re-numbering, but it would probably be good to check once more

**Comment by SimonBiggs** [comment]
> IMHO, Mention of the imports and inspiration like pydicom and pylinac, yes.
> Additional consumers like OnkoDICOM maybe not, unless they have already asked for how to reference us.

@sjswerdloff or @rembishj would either of you be in a position to add a paragraph (or sentence) to this effect?

Let's focus on adding those packages within the medical physics ecosystem that we directly import:

* pydicom, pynetdicom, pylibjpeg-libjpeg (https://doi.org/10.5281/zenodo.6394735)
* dicompyler-core (https://dx.doi.org/10.5281/zenodo.3236628)
* pylinac (https://doi.org/10.1016/j.ejmp.2016.07.122)

**Comment by SimonBiggs** [comment]
> @SimonBiggs I double-checked the affiliations after re-numbering, but it would probably be good to check once more

LGTM :)

**Comment by SimonBiggs** [comment]
> Can't wait to release the next generation =)

I'm keen for its release :slightly_smiling_face:. Remember to let me know if you want to include auto-contouring capacity :)

File: paper.md
Comment (Matthew-Jennings) [inline paper.md:71]: Suggest a reword that at least acknowledges non-radiotherapy medical physics. Also suggest alternative to "delivery methods", since this might imply innovation beam _delivery_ techniques only when there have been significant innovations relevant to PyMedPhys in simulation, planning, guidance, etc.

Perhaps the paragraph could be replaced with something like:

_"Medical radiation applications are subject to fast-paced technological advancements. This is particularly true in the field of radiation oncology, where the implementation of increasingly sophisticated technologies requires increasingly complex processes to maintain the improving standards of care. To address this challenge, medical physicists are increasingly developing in-house software tools to improve the quality, safety and efficiency of clinical tasks. However, in many cases, such development is limited to the physicist's own clinic. Consequently, many similar tools that could otherwise be shared are instead "reinvented" on a routine basis. Moreover, individual clinics often lack the personnel resources to incorporate simple good development practices such as peer-review."_

Just a draft that could be tweaked. Noticed I've removed the AAPM comment. That could be reintroduced, but I would suggest that such a claim needs a reference to be included.

File: paper.md
Comment (Matthew-Jennings) [inline paper.md:82]: Sometimes I wonder whether we should be marketing this to "medical physicists" or instead simply marketing PyMedPhys as a "medical physics toolkit". There is a subtle difference - the latter makes no implicit claim that the tasks traditionally considered within the scope of medical physics are/should only performed by medical physic**ists**.

**Comment by SimonBiggs** [comment]
@rembishj, if you're okay with it, I'll let you make the judgement call regarding @Matthew-Jennings comments :slightly_smiling_face:

**Comment by rembishj** [comment]
> @rembishj, if you're okay with it, I'll let you make the judgement call regarding @Matthew-Jennings comments 🙂

I think those seem to be very valid points, and I'm okay with adjusting things to fit that!

File: paper.md
Comment (crcrewso) [reply paper.md:82]: I would strongly support that language change! It feels more inclusive, though this might be a personal opinion since I am a physicist but not a Medical Physicist.

**Comment by Matthew-Jennings** [comment]
Hi all. I've created a PR - for merge into this PR - with suggested changes: #1641

**Review by Matthew-Jennings — APPROVED** [review]

**Comment by Matthew-Jennings** [comment]
The Windows doctest will likely pass after #1648 and #1559 land. If there are no objections, I think we can merge after then. @rembishj, would you like to proceed with submission to JOSS? ? I can assist if needed.

**Comment by Matthew-Jennings** [comment]
I hate to be "that guy", but I would really appreciate being listed as second author for this paper. My career still involves an academic component where these things perhaps matter a little more. I'd also cheekily point out that I've been contributing/maintaining the project for the longest period after @SimonBiggs by a significant margin, albeit with breaks. I also have the second most contributions as measured by code line changes & commits (admittedly a weak indicator, but relevant nonetheless). Combined with my textual contribution to the paper, I think this is a reasonable request.

**Comment by SimonBiggs** [comment]
Hi @Matthew-Jennings :slightly_smiling_face:,

You most certainly have made many contributions to PyMedPhys.

I would like to flag that @sjswerdloff has also poured in an invaluable number of code reviews and feedback well outnumbering his other contribution types. And, as you can attest, this review work can be one of the harder components of being a PyMedPhys maintainer.

> I also have the second most contributions as measured by code line changes & commits (admittedly a weak indicator, but relevant nonetheless).

If we're pulling in stats there might be value in pulling in the following tool as well?

https://github.com/trialspark/code-review-stats

Might you be able to run that?

**Comment by Matthew-Jennings** [comment]
> Hi @Matthew-Jennings 🙂,
> 
> You most certainly have made many many contributions to PyMedPhys.
> 
> > I also have the second most contributions as measured by code line changes & commits (admittedly a weak indicator, but relevant nonetheless).
> 
> I would like to flag that @sjswerdloff has also poured in an invaluable number of code reviews and feedback well outnumbering his other contribution types. And, as you can attest, this review work can be one of the harder components of being a PyMedPhys maintainer.
> 
> If we're pulling in stats there might be value in pulling in the following tool as well?
> 
> https://github.com/trialspark/code-review-stats
> 
> Might you be able to run that?
> 
> Or maybe the following tool?
> 
> https://github.com/change-metrics/monocle
> 
> Or maybe you can find a better one. But anyway, if stats are desired to be used as a justification here, then a review count needs to be included...

I added statistics as a supplementary point only. There are many relevant, uncaptured metrics on both sides (consider, e.g., the only-partial relocation from GitLab, the relative utility and ongoing support of pymedphys tools, the relative utility of said reviews, etc.). I figure this is a judgement call that probably only you can make.

I'm not going to push the point any further. It was a simple request for which, if the answer isn't obvious, I have no desire to argue.

**Comment by SimonBiggs** [comment]
> I figure this is a judgement call that probably only you can make.

Fair enough. When making the decision initially, not wanting to put a weighting to the various contributions, I opted to go with the ordering on the "our-team page" which acknowledges the ongoing commitment of those who are PyMedPhys maintainers.

**Comment by Matthew-Jennings** [comment]
The "our-team page" is a snapshot of activity now. This paper should represent contributions over PyMedPhys' history with no bias toward the present day. It's the best academic recognition of this work that we (will) have.

**Comment by SimonBiggs** [comment]
> This paper should represent contributions over PyMedPhys' history with no bias toward the present day.

When looking over the history of PyMedPhys, you both have provided immense support. And I am struggling to justify putting one over the other when judging by scope, size, and impact of contributions. 

I want you to understand @Matthew-Jennings, that this decision is not underselling your contributions, but instead acknowledging the sheer magnitude of work that @sjswerdloff has undergone over these last two years. So, as an imperfect measure to help justify my decision to leave them as is, here is a snapshot of the total number of PRs approved and merged under a review + approval system across both the GitHub and GitLab days of PyMedPhys:

Matt Jennings:
[32 @ GitLab](https://gitlab.com/pymedphys/pymedphys/-/merge_requests?approved_by_usernames%5B%5D=centrus&scope=all&state=merged) + [139 @ GitHub](https://github.com/pymedphys/pymedphys/pulls?q=is%3Apr+reviewed-by%3AMatthew-Jennings+)

Stuart Swerdloff:
[188 @ GitHub](https://github.com/pymedphys/pymedphys/pulls?q=is%3Apr+reviewed-by%3Asjswerdloff+)

**Comment by sjswerdloff** [comment]
> > This paper should represent contributions over PyMedPhys' history with no bias toward the present day.
> 
> When looking over the history of PyMedPhys, you both have provided immense support. And I am struggling to justify putting one over the other when judging by scope, size, and impact of contributions.
> 
> I want you to understand @Matthew-Jennings, that this decision is not underselling your contributions, but instead acknowledging the sheer magnitude of work that @sjswerdloff has undergone over these last two years. So, as an imperfect measure, to help justify my decision to leave them as is, here is a snapshot of the total number of PRs approved and merged under a review + approval system across both the GitHub and GitLab days of PyMedPhys:
> 
> Matt Jennings: [32 @ GitLab](https://gitlab.com/pymedphys/pymedphys/-/merge_requests?approved_by_usernames%5B%5D=centrus&scope=all&state=merged) + [139 @ GitHub](https://github.com/pymedphys/pymedphys/pulls?q=is%3Apr+reviewed-by%3AMatthew-Jennings+)
> 
> Stuart Swerdloff: [188 @ GitHub](https://github.com/pymedphys/pymedphys/pulls?q=is%3Apr+reviewed-by%3Asjswerdloff+)

While I believe in gathering metrics, I am hesitant to put too much faith in them.
I'd feel a lot better about being further down the list in terms of authorship.

**Comment by SimonBiggs** [comment]
> I'd feel a lot better about being further down the list in terms of authorship.

Fair enough :) @sjswerdloff would you be able to make the swap with @Matthew-Jennings?

**Comment by sjswerdloff** [comment]
> > I'd feel a lot better about being further down the list in terms of authorship.
> 
> Fair enough :) @sjswerdloff would you be able to make the swap with @Matthew-Jennings?

@Matthew-Jennings would you please double check I didn't mangle anything...

**Comment by SimonBiggs** [comment]
Before this is submitted, we have had some others join the community through their pull requests while it has been in writing. Would @guanfada, @crcrewso and @nlambriICH also be interested in being included as authors?

**Comment by guanfada** [comment]
@SimonBiggs Thanks a lot for including me! That would be my great honor to be part of this work. Thanks so much! 
I agree that we do need at least a publication as the major reference when others cite PyMedPhys.

**Comment by SimonBiggs** [comment]
> That would be my great honor to be part of this work.

Brilliant :)

Would you be able to go to Files changed:

image

Scroll down to the paper.md and click "Edit file":

image

Then, add your name under authors at the following point:

image

Include your affiliation here:

image

Then fix up the affiliation numbering.

Then lastly, review the text of the paper, provide any suggestions you may have, and then leave your approval on this PR. :slightly_smiling_face:

**Comment by SimonBiggs** [comment]
And @cedricbelanger, would you also be interested in being an author?

**Review by guanfada — APPROVED** [review]

**Comment by guanfada** [comment]
@SimonBiggs Where is the draft of the paper manuscript, so I can take a look at it? Thanks!

**Comment by SimonBiggs** [comment]
> Where is the draft of the paper manuscript, so I can take a look at it?

Yup, click "Checks":

image

Paper:

image

Then download the zip that contains the pdf over at:

image

**Review by guanfada — APPROVED** [review]

**Comment by guanfada** [comment]
@SimonBiggs I just downloaded the manuscript. Thanks!

**Comment by guanfada** [comment]
@SimonBiggs I have a co-authored paper manuscript that also used gamma module. We just got the journal reviewers' criticisms. Hopefully, that paper can be published soon, and then I can cite it in the Statement of Need of this JOSS paper. Let's see.

**Comment by Matthew-Jennings** [comment]
> While I believe in gathering metrics, I am hesitant to put too much faith in them.
> I'd feel a lot better about being further down the list in terms of authorship.

Thank you very much, @sjswerdloff.

> @Matthew-Jennings would you please double check I didn't mangle anything...

LGTM :)

**Comment by SimonBiggs** [comment]
> Hopefully, that paper can be published soon, and then I can cite it in the Statement of Need of this JOSS paper. Let's see.

That reminds me, @Matthew-Jennings, while this PR has been open there has actually been a reasonable number of new PyMedPhys citations. Might be worth doing a Google scholar check to make sure this paper has captured them before we submit to JOSS?

**Comment by Matthew-Jennings** [comment]
> > Hopefully, that paper can be published soon, and then I can cite it in the Statement of Need of this JOSS paper. Let's see.
> 
> That reminds me, @Matthew-Jennings, while this PR has been open there has actually been a reasonable number of new PyMedPhys citations. Might be worth doing a Google scholar check to make sure this paper has captured them before we submit to JOSS?

Oh, were the refs [here](https://github.com/pymedphys/pymedphys/blob/202dea82f68a8e5d26f28dfbae7c25e3507cb7c5/paper.md?plain=1#L92) intended to be all-inclusive? I (perhaps mistakenly) thought it was just a snapshot of clear examples.

**Comment by crcrewso** [comment]
> Before this is submitted, we have had some others join the community through their pull requests while it has been in writing. Would @guanfada, @crcrewso and @nlambriICH also be interested in being included as authors?

If the group is offering me the honor of being listed as an author I'd happily accept. Should I submit a pr with my affiliation?

**Comment by SimonBiggs** [comment]
> If the group is offering me the honor of being listed as an author I'd happily accept. Should I submit a pr with my affiliation?

Most certainly, yes please :)

**Comment by nlambriICH** [comment]
> Before this is submitted, we have had some others join the community through their pull requests while it has been in writing. Would @guanfada, @crcrewso and @nlambriICH also be interested in being included as authors?

@SimonBiggs I will certainly be heppy to be on the list of authors if you think I deseverd it with my contributions :)

**Comment by SimonBiggs** [comment]
> Thank you very much, @sjswerdloff.

What do you reckon @Matthew-Jennings, when we all meet up, drinks are shouted in author order? :)

File: paper.md
Comment (SimonBiggs) [inline paper.md:34]: @crcrewso, would you be able to bump yourself up one. @msobolewski there is taking the final spot on the author list as an overarching project lead from the very early days (before it was even called PyMedPhys :slightly_smiling_face:)

**Comment by cedricbelanger** [comment]
Hi @SimonBiggs !

Thank you for inviting me to this work. It will be great for the community to have a paper published. I have a paper under revision which used pyMedPhys gamma module, I would like to know the reference when this work is submitted such that I can update it before (hopefully) publication. As of my contribution for authorship, I feel that Acknowledgements would be more appropriate :). I look forward to read this paper!

Best regards,

Cédric

**Comment by SimonBiggs** [comment]
> I feel that Acknowledgements would be more appropriate

Okay. Would you be okay with writing the appropriate acknowledgment in by editing this pull request?

But also, just so you know, here are the authorship requirements:

https://joss.readthedocs.io/en/latest/submitting.html#authorship

And being a part of the PyMedPhys community through repository contributions, being willing to provide support around those contributions into the future, as well as being likely to continue to contribute into the future, have been the measuring stick we've gone by this far for authorship.

I suspect you may fit those requirements... But if you'd prefer acknowledgement instead that is completely understandable.

Cheers,
Simon

File: paper.md
Comment (crcrewso) [reply paper.md:32]: Absolutely, that's a big faux pas on my part. Sorry @msobolewski for missing that one.

Comment (SimonBiggs) [reply paper.md:32]: All good :) easy fix :)

**Comment by matthewdeancooper** [comment]
Hey @SimonBiggs and everyone else :), wondering if i should add myself as an author or in the acknowledgements section and where to order myself as such?

Cheers

**Comment by SimonBiggs** [comment]
Hi Matt, yes please, add yourself as an author (as long as it's okay with your work). Can you put your name just below Jacob McAloney?

**Comment by sjswerdloff** [comment]
@SimonBiggs How does one adjust the links in the references to the actual repository (the pymedphys repo on GitHub) and archive (on PyPI ?) in the section that shows up on the left hand side with the following text ?
```
Software
• Review 
• Repository 
• Archive
```
Right now it's pointing to JOSS itself.
I looked through the files in the PR, and I don't see that text

**Comment by SimonBiggs** [comment]
> How does one adjust the links in the references to the actual repository

I believe that side of things gets sorted out once we begin the submission process with JOSS. That rendered paper is more of a pre-print set up to validate that we've got our side of things sorted out.

**Comment by SimonBiggs** [comment]
> The Windows doctest will likely pass after #1648 and #1559 land. If there are no objections, I think we can merge after then. @rembishj, would you like to proceed with submission to JOSS? ? I can assist if needed.

No need to wait for the Windows doctests to pass. @Matthew-Jennings if you're interested in spearheading this I'm happy for you to make the JOSS submission if you'd like? @rembishj successfully got the ball rolling, but if you're keen to carry it across the line I think that would work well.

**Comment by Matthew-Jennings** [comment]
> > The Windows doctest will likely pass after #1648 and #1559 land. If there are no objections, I think we can merge after then. @rembishj, would you like to proceed with submission to JOSS? ? I can assist if needed.
> 
> No need to wait for the Windows doctests to pass. @Matthew-Jennings if you're interested in spearheading this I'm happy for you to make the JOSS submission if you'd like? @rembishj successfully got the ball rolling, but if you're keen to carry it across the line I think that would work well.

Will do. Might be next week at this stage but I'll try for this week.

**Comment by SimonBiggs** [comment]
> Will do. Might be next week at this stage but I'll try for this week.

@matthewdeancooper, if you'd like your name on the author list best do it sooner rather than later. @cedricbelanger if you'd be keen for an acknowledgement beyond what is already in the paper, best get that PR in sooner rather than later.

**Comment by Matthew-Jennings** [comment]
> > Will do. Might be next week at this stage but I'll try for this week.
> 
> @matthewdeancooper, if you'd like your name on the author list best do it sooner rather than later. @cedricbelanger if you'd be keen for an acknowledgement beyond what is already in the paper, best get that PR in sooner rather than later.

@matthewdeancooper & @cedricbelanger: perhaps even if you just give me an indication of an expected timeframe to make changes, I will hold off until then. :)

**Comment by SimonBiggs** [comment]
Hi @Matthew-Jennings,

For a few key reasons I'd like to get the ball rolling on this one. Would you be in a position to make the submission to JOSS? Otherwise I can jump on submitting it to JOSS tomorrow.

**Comment by Matthew-Jennings** [comment]
> Hi @Matthew-Jennings,
> 
> For a few key reasons I'd like to get the ball rolling on this one. Would you be in a position to make the submission to JOSS? Otherwise I can jump on submitting it to JOSS tomorrow.

Yes, I can submit. Probably not until tomorrow either, though. =/

**Comment by Matthew-Jennings** [comment]
Shall we merge then? Or leave in this branch and notify JOSS accordingly (they accept both options)

**Comment by SimonBiggs** [comment]
> Shall we merge then? Or leave in this branch and notify JOSS accordingly? (they accept both options)

Let's merge, but I assume we'll still be free to make a few tweaks on the paper before it's all finalised?

**Comment by guanfada** [comment]
@SimonBiggs @Matthew-Jennings I tried to add a new citation in the paper.bib, but the "edit" option is grayed out.

**Comment by SimonBiggs** [comment]
> Is the parent project glad to be in the "same family"?

Most definitely 😁. And if you have any ideas on how the projects can find further synergies we are all ears 🙂👂☺️

**Pull request merged**
